### ISSUE TITLE
Beautifying html replaces > with ]

### ISSUE THREAD

**Issue by filipesilva** [prettydiff/prettydiff#100]
also in Glavin001/atom-beautify#365

Beautifying

    <input value="1" checked="number > -1"/>

gives out

    <input checked="number ] -1" value="1"/>

This is relevant since comparisons in html are common in angular.

**Comment by prettydiff** [comment]
Won't fix according to comment: issuecomment-105261191

**Comment by prettydiff** [comment]
Reopening with a better code sample from an AngularJS template.

**Comment by prettydiff** [comment]
<input  id="{{group.id}}" type="checkbox" value="{{group.id}}" ng-checked="campaign.groupRecipients.indexOf(group.id) > -1" ng-click="toggleSelection(group.id)" />

**Comment by mattanbi** [comment]
Hi, will this issue be fixed in the next version?

**Comment by prettydiff** [comment]
Yes, I am rewriting my markup parser to solve this and other unrelated problems. I am half way through right now and will let everyone one the moment I have something worthy of review.

**Comment by prettydiff** [comment]
A beta version of the new markup parser is up. This version is feature complete, but not fully tested against edge cases.

http://prettydiff.com/ignore/testlocation/prettydiff.com.xhtml

Please test this out and try to break it. If you encounter any issues please post them here.  This is likely to be buggy because it is completely new parser written from scratch in the past two weeks.

**Comment by filipesilva** [comment]
Will do. Thanks for your hard work in this!

**Comment by filipesilva** [comment]
Is it working right now? I get some errors
```
Line 9486 at column 45 of http://prettydiff.com/ignore/testlocation/prettydiff.jsdom__error @ dom.js:149
dom.js:150 Help a guy out and open a bug: https://github.com/prettydiff/prettydiff/issues

Uncaught ReferenceError: content is not defined
dom.js:148 OPTIONS http://prettydiff.com:8000/error/ net::ERR_CONNECTION_REFUSED
```

**Comment by prettydiff** [comment]
I am so sorry about that. This particular error was due to a missing comma.  The code is now fully linted. Please try again.

**Comment by filipesilva** [comment]
The problem I reported is not present anymore. I ran a production file full of angular markup and properties and it turned out fine.

I noticed that a warning was displayed on `<img src='something.png'>` due to it being unclosed, but as far as I know it is [valid html5 syntax](http://stackoverflow.com/a/14860596/2116927). I'm perfectly happy with having warnings in it and self-closing them, just wondered if it was intended behavior.

**Comment by prettydiff** [comment]
I am using Ace Editor in the HTML UI. If the warning is coming in the form a red or yellow icon in the gutters next to the line numbers then it is a warning from Ace. My only contribution to this is to ensure I am auto detecting the input code correctly and communicating the language preference to Ace properly.

Otherwise I treat certain tags, by name, as a singleton (not needing an end tag) if the language is HTML opposed to XML or something unknown. The really tricky one is that HTML5 does not require a closing tag for list items (</li>), but a list item may contain large tag structures. I try to account for this as well.

**Comment by filipesilva** [comment]
Yeah it was a gutter warning, so it's not part of prettydiff. I did not know that about `li `, and that sounds pretty messed up!

I'm happy to close this issue if you are, the new version fixes it.

**Comment by prettydiff** [comment]
Great, I will close it just as soon as I move this change to production.

**Comment by kaizer200** [comment]
Hi,
found a bug when performing **beautification on markup**.
tested xml:
```<ttt> <a    at=3></a>
    <b dd=1 cc=2></b>
   </ttt>```
result xml:
```<ttt><a at=3></a>
    <b 2 cc=  dd=1 ></b>
</ttt>```
check out the <b 2 in the result ...
I think that there was such an issue in the past which you've solved

**Comment by filipesilva** [comment]
Don't xml attributes always need to be enclosed with quotes?

**Comment by kaizer200** [comment]
1) you are right for attributes values, but still - beautification should mess up you're xml content.
2) more impotently:
 just checked per your comment, and with quotes, the behavior is **false** -
(e.g: cc= "2"   > gets messed also)
3) This is a regression (at least from v. 1.11.13)

**Comment by prettydiff** [comment]
@kaizer200 would you be able to provide an XML tag example that demonstrates the failure?

**Comment by kaizer200** [comment]
sure @prettydiff  ,
(I've placed it in a comment before my last one)
here you go (there is space/s between cc= and "2":
```<ttt><a at="3"></a> <b dd="1" cc= "2"></b> </ttt>```

The result is 
```<ttt> <a at="3"></a>  <b "2" cc=  dd="1"></b></ttt>```

**Comment by prettydiff** [comment]
@kaizer200 yeah, I am using unquoted spaces as an attribute terminator. As a result the parser things `b` tag has 3 attributes instead of two. Is this really a bug in the parser though?

**Comment by kaizer200** [comment]
This is a bug indeed:
1) XML permits such a case. Plus - I've also checked with chrome's xml parser.
When beautify such a valid case, it transforms a valid XML into an invalid XML - and this is a substantial bug... 
2) check out our correspondence in prettyDiff google group, topic - " xml formating - wrapping line" - my post from May 5th (point 3)  and onward (i can attach if you like). You've entered a fix for this issue in version 1.11.10 or 1.11.11.

**Comment by filipesilva** [comment]
As a reference, in the [current](http://prettydiff.com/) version I inputted @kaizer200's code

```
<ttt><a at="3"></a> <b dd="1" cc= "2"></b> </ttt>
```

and got 
```
<ttt>
    <a at="3"></a>
    <b cc="2" dd="1"></b>
</ttt>
```

I suppose the current parser ignores whitespace after `=`?

**Comment by prettydiff** [comment]
I just added support for this edge case of 1 or more white space characters occurring between the `=` and a quoted value.  If the value is not quoted then they will remain as separate attributes.

**Comment by kaizer200** [comment]
Thanks,
Could you comment at which milestone you relase?

**Comment by prettydiff** [comment]
Fixed as v1.12.0

**Comment by filipesilva** [comment]
Awesome :D
